sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -101,4 +102,30 @@ ng_netreg_entry_t *ng_netreg_getnext(ng_netreg_entry_t *entry)
     return entry;
 }
 
+int ng_netreg_calculate_csum(ng_nettype_t pseudo_hdr_type,
+                             ng_nettype_t csum_hdr_type,
+                             ng_pktsnip_t *pkt)
+{
+    switch (csum_hdr_type) {
+#ifdef MODULE_NG_ICMPV6
```

Comment:
I guess you cannot define empty-body functions, when these functions need to return something else then void, can you? :/ This would nullify my idea, so I think...